lisp/ess-custom.el

```diff
@@ -646,211 +646,345 @@ regardless of where in the line point is when the TAB command is used."
 nil means to use R/S indentation.")
 (make-variable-buffer-local 'ess-indent-line-function)
 
-(defvar ess-indent-level 2
+(defvar ess-indent-level 4
   "Indentation of S statements with respect to containing block.")
 
-(defvar ess-brace-imaginary-offset 0
-  "Imagined indentation of an open brace following a statement.")
+(defvar ess-offset-statement '(t)
+  "Indentation for statement. A statement is usually declared
+with braces but also with anonymous parentheses.
 
-(defvar ess-brace-offset 0
-  "Extra indentation for open braces.
-Compares with other text in same context.")
+If a number N, blocks are indented relative to the opening
+parenthesis of the closest function call:
 
-(defvar ess-first-continued-statement-offset 0
+  {
+      fun_call(parameter = {
+                   stuff
+               }, {
+                   stuff
+               })
+
+      lapply(data, function(x) {
+                 body
+             })
+  }
+
+In this case, the value types of `ess-offset-arguments' and
+`ess-offset-arguments-newline' are taken into account for
+consistency.
+
+
+If a list of the form '(N) where N is a number, blocks are
+indented at the previous line indentation + N characters:
+
+  {
+      fun_call(parameter = {
+          stuff
+      }, {
+          stuff
+      })
+
+      lapply(data, function(x) {
+          body
+      })
+  }
+
+
+If nil, blocks are indented from the opening delimiter:
+
+  {
+      fun_call(parameter = {
+                               stuff
+                           }, {
+                                  stuff
+                              })
+
+      lapply(data, function(x) {
+                                   body
+                               })
+  }
+
+
+You can refer to ess-indent-level by setting this parameter to t
+or '(t) instead of N or '(N).
+")
+
+(defvar ess-offset-statement-opening 0
+  "Extra indentation for delimiters opening a statement. Only has
+an effect when the statement is not part of an expression.")
+
+(defvar ess-offset-statement-closing 0
+  "Extra indentation for delimiters closing a statement. Only has
+an effect when the statement is not part of an expression.")
+
+(defvar ess-offset-statement-first-continued 0
   "Extra indentation for the first new line continuing an expression.
 If you set this to non-zero value you might want to set
-`ess-continued-statement-offset' to zero.")
+`ess-offset-statement-continued' to zero.
 
-(defvar ess-continued-statement-offset 2
-  "Extra indent for lines not starting new statements.")
+You can refer to ess-indent-level by setting this parameter to t.")
 
-(defvar ess-continued-brace-offset 0
-  "Extra indent for substatements that start with open-braces.
-This is in addition to ess-continued-statement-offset.")
+(defvar ess-offset-statement-continued 2
+  "Extra indent for lines not starting new statements.
+You can refer to ess-indent-level by setting this parameter to t.")
+
+(defvar ess-offset-statement-continued-opening 0
```

Comment:
> I don't think we need it, it just was there before. Should I remove it then?

Yes. Please. Let's try to reduce variables to the minimum and then we see if we need to add anything. Sorry to bother you on this, but (as you surely noticed) no one seems to have any clue anymore what is going on with indentation. it just kept growing wildly over the years. This is the right time to clean the house I think.